src/panoramic/cli/pano_model.py

```diff
@@ -56,6 +56,96 @@ def __eq__(self, o: object) -> bool:
         return self.to_dict() == o.to_dict()
 
 
+class PanoField(Actionable):
+    """Field definition."""
+
+    API_VERSION = 'v1'
+
+    def __init__(
+        self,
+        *,
+        slug: str,
+        group: str,
+        display_name: str,
+        data_type: str,
+        field_type: str,
+        description: Optional[str],
+        data_source: Optional[str],
+        calculation: Optional[str],
+        display_format: Optional[str],
+        aggregation: Optional[Dict[str, Any]],
+        package: Optional[str] = None,
+        file_name: Optional[str] = None,
+    ):
+        self.slug = slug
+        self.group = group
+        self.display_name = display_name
+        self.data_type = data_type
+        self.field_type = field_type
+        self.description = description
+        self.data_source = data_source
+        self.calculation = calculation
+        self.display_format = display_format
+        self.aggregation = aggregation
+        self.package = package
+        self.file_name = file_name
+
+    def to_dict(self) -> Dict[str, Any]:
+        data: Dict[str, Any] = {
+            'api_version': self.API_VERSION,
+            'slug': self.slug,
+            'group': self.group,
+            'display_name': self.display_name,
+            'data_type': self.data_type,
+            'field_type': self.field_type,
+            # data_source is not persisted to YAML
+        }
+
+        if self.description is not None:
+            data['description'] = self.description
+        if self.calculation is not None:
+            data['calculation'] = self.calculation
+        if self.aggregation is not None:
+            data['aggregation'] = self.aggregation
+        if self.display_format is not None:
+            data['display_format'] = self.display_format
+
+        return data
+
+    @classmethod
+    def from_dict(cls, inputs: Dict[str, Any]) -> 'PanoField':
+        return cls(
+            slug=inputs['slug'],
+            group=inputs['group'],
+            display_name=inputs['display_name'],
+            data_type=inputs['data_type'],
+            field_type=inputs['field_type'],
+            description=inputs.get('description'),
+            data_source=inputs.get('data_source'),
+            aggregation=inputs.get('aggregation'),
+            calculation=inputs.get('calculation'),
+            display_format=inputs.get('display_format'),
+            package=inputs.get('package'),
+            file_name=inputs.get('file_name'),
+        )
+
+    @property
+    def id(self) -> Union[Tuple[str], Tuple[str, str]]:
+        if self.data_source is not None:
+            return (self.data_source, self.slug)
+        else:
+            return (self.slug,)
+
+    def __hash__(self) -> int:
+        return hash(self.id)
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, PanoField):
+            return False
+
+        return self.id == o.id
```

Comment:
We should compare all values, not just id here

Reply:
Fixed in 81452a00f02d4ef2d193499b8ebbd57901575602.